pysteps/downscaling/rainfarm.py

```diff
@@ -119,8 +173,18 @@ def downscale(precip, ds_factor, alpha=None, threshold=None, return_alpha=False)
     tophat = ((mx**2 + my**2) <= rad**2).astype(float)
     tophat /= tophat.sum()
 
-    P_agg = _balanced_spatial_average(P_u, tophat)
```

Comment:
From what I understand, you could keep the calls to the `_balanced_spatial_average` function before the if block and simply call `_smoothconv` if `smooth` is True. This would avoid some code duplication.